pkg/cmd/flags.go

```diff
@@ -0,0 +1,172 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/fastly/cli/pkg/api"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/go-fastly/v3/fastly"
+	"github.com/fastly/kingpin"
+)
+
+// ServiceVersionFlagOpts enables easy configuration of the --version flag
+// defined via the NewServiceVersionFlag constructor.
+//
+// NOTE: The reason we define an 'optional' field rather than a 'required'
+// field is because 99% of the use cases where --version is defined the flag
+// will be required, and so we cater for the common case. Meaning only those
+// subcommands that have --version as optional will need to set that field.
+type ServiceVersionFlagOpts struct {
+	Dst      *string
+	Optional bool
+	Action   kingpin.Action
+}
+
+// NewServiceVersionFlag defines a --version flag that accepts multiple values
+// such as 'latest', 'active' and numerical values which are then converted
+// into the appropriate service version.
+func (b Base) NewServiceVersionFlag(opts ServiceVersionFlagOpts, args ...string) {
+	clause := b.CmdClause.Flag("version", "Number of service version, 'latest', 'active' or 'editable'")
```

Comment:
This sounds a bit like "Number of [service version, latest, active or editable]", not "[Number of service version], latest, active or editable".  Resolve that ambiguity:

```suggestion
	clause := b.CmdClause.Flag("version", "'latest', 'active', 'editable', or the number of a specific service version")
```